pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -126,6 +127,10 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 		return nil, err
 	}
 
+	// Update query stats.
+	queryStats := stats.FromContext(ctx)
+	queryStats.AddSplitQueries(uint32(len(splitReqs)))
```

Comment:
I think it's better to check the actual number of downstream requests after the cache has been looked up. Basically, I would report `execReqs`, which is the actual number of split queries we'll run.